src/coreclr/nativeaot/Common/src/Internal/Runtime/MethodTable.cs

```diff
@@ -1414,8 +1414,8 @@ public uint GetFieldOffset(EETypeField eField)
                 Debug.Assert((rareFlags & EETypeRareFlags.IsDynamicTypeWithThreadStatics) != 0);
                 return cbOffset;
             }
-            if ((rareFlags & EETypeRareFlags.IsDynamicTypeWithThreadStatics) != 0)
-                cbOffset += 4;
+            //if ((rareFlags & EETypeRareFlags.IsDynamicTypeWithThreadStatics) != 0)
```

Comment:
Is this intentional? We should remove the code if we don't need it.

Reply:
It looks unintentional so I didn't want to remove straight remove it /cc @MichalStrehovsky

Reply:
What's the value in commenting out?

Reply:
Leaving it commented out is fine. This is code for reading a variable length data structure and this was the last field. The code is unreachable (we would run into the assert on the line below); it's just futureproofing for when we add a new field since these are all paired (read the field if that's the field we want, or skip over the field if it's not the field we want).

Reply:
Alternatively, we could just disable the diagnostic for this whole method.

Reply:
OK, just removed the code